app/services/experiment_control_service.rb

```diff
@@ -0,0 +1,94 @@
+class InvalidExperiment < StandardError; end
+
+class ExperimentControlService
+  class << self
+    def start_current_patient_experiment(name, days_til_start, days_til_end)
+      raise ArgumentError, "Start date must be before end date" if days_til_end < days_til_start
+
+      existing_experiment = Experimentation::Experiment.where(experiment_type: "current_patient_reminder", state: ["selecting", "live"])
+      raise InvalidExperiment, "A current patient experiment is already in progress" if existing_experiment.any?
+
+      experiment = Experimentation::Experiment.find_by!(name: name, experiment_type: "current_patient_reminder")
+      experiment_start = days_til_start.days.from_now.beginning_of_day
+      experiment_end = days_til_end.days.from_now.end_of_day
+
+      experiment.state_selecting!
+
+      experiment_patients = patient_pool
+        .joins(:appointments)
+        .where("appointments.status = ?", "scheduled")
+        .where("appointments.scheduled_date BETWEEN ? AND ?", experiment_start, experiment_end)
+        .order(Arel.sql("random()"))
+
+      experiment_patients.each do |patient|
+        group = experiment.group_for(patient.id)
+        appointment = patient.appointments
+          .where(status: "scheduled")
+          .where("appointments.scheduled_date BETWEEN ? AND ?", experiment_start, experiment_end).first
```

Comment:
Why just the first? If we get multiple w/I the experiment time, wouldn't we want reminders for all?

Reply:
That's a valid point. I'd been told a patient should only have one scheduled appointment, but I should still code on the assumption that that might not be true.